lib/api/core/plugins/pluginsManager.js

```diff
@@ -876,19 +887,30 @@ function triggerPipes(pipes, event, data) {
 }
 
 /**
- * For a specific event, return the corresponding wildcard
+ * For a specific event, return the correspondings wildcards
  * @example
- *  getWildcardEvent('data:create') // return 'data:*'
- * @param {string} event
- * @returns {String} wildcard event
+ *  getWildcardEvents('data:create') // return ['data:*']
+ *  getWildcardEvents('data:beforeCreate') // return ['data:*', 'data:before*']
+ * @param {String} event
+ * @returns {Array<String>} wildcard events
  */
-function getWildcardEvent (event) {
-  const indexDelimiter = event.indexOf(':');
-  if (indexDelimiter !== 1) {
-    return event.substring(0, indexDelimiter+1) + '*';
+function getWildcardEvents (event) {
+  const [scope, name] = event.split(':');
+
+  if (!name) {
+    return null;
   }
 
-  return null;
+  const wildcardEvents = ['*:*', `${scope}:*`];
```

Comment:
I disagree with the event `*:*` because:
- we can have an arbitrary number of separators (for instance we have a [core:auth:strategyAdded](https://docs.kuzzle.io/plugins/1/events/core-auth-strategy-added/) event)
- I think that `*` would be much simpler


```suggestion
  const wildcardEvents = ['*', `${scope}:*`];
```

Reply:
As we discuss with @thomasarbona, it could be nice to handle `wildcard` pattern directly as regex. So we do not need to define rules for patterns and use regex syntax.

Reply:
Veto: regexp are slow, and the number and frequency of events triggered make them a critical part of the code.

Reply:
You're right I didn't get that this action appends at funnel level. Ignore my previous comment :sweat_smile: